WordPress/Classes/Models/Blog+Media.swift

```diff
@@ -0,0 +1,37 @@
+import Foundation
+
+extension Blog {
+
+    /// Get the number of items in a blog media library that are of a certain type.
+    ///
+    /// - Parameter mediaTypes: set of media type values to be considered in the counting.
+    /// - Returns: Number of media assets matching the criteria.
+    @objc(mediaLibraryCountForTypes:)
+    func mediaLibraryCount(types mediaTypes: NSSet) -> Int {
+        guard let context = managedObjectContext else {
+            return 0
+        }
+
+        var count: Int?
+        context.performAndWait {
+            var predicate = NSPredicate(format: "blog == %@", self)
+
+            if mediaTypes.count > 0 {
+                let types = mediaTypes
+                    .map { obj in
+                        guard let rawValue = (obj as? NSNumber)?.uintValue,
+                              let type = MediaType(rawValue: rawValue) else {
+                            fatalError("Can't convert \(obj) to MediaType")
```

Comment:
As you know, I'm always pedantic when it comes to `fatalError` or `preconditionFailure` usages, preferring to bubble up errors for the app to use instead of crashing.

The rule of thumb we agree with in regards to crashing the app is "only in case of developer error," and this seems like a good instance of developer error: somehow passing one or more values to the set that don't convert to `MediaType`. ✅

Moreover, this method is called only in one place.

If we'll ever get to lifting the call to the Swift layer, then we can change the input parameter type and make this inconsistent state unrepresentable.